crates/doc/src/reduce/strategy.rs

```diff
@@ -301,6 +309,54 @@ impl Strategy {
         }
     }
 
+    fn json_schema_merge<'alloc, L: AsNode, R: AsNode>(
+        cur: Cursor<'alloc, '_, '_, '_, '_, L, R>,
+    ) -> Result<HeapNode<'alloc>> {
+        let Cursor {
+            tape,
+            loc,
+            full: _,
+            lhs,
+            rhs,
+            alloc,
+        } = cur;
+
+        let (lhs, rhs) = (lhs.into_heap_node(alloc), rhs.into_heap_node(alloc));
+
+        // Precompute this up here because we're going to consume `rhs` later on
+        let rhs_tape_len = count_nodes_heap(&rhs);
```

Comment:
nit: i think you can just update *tape immediately. You don't have any nested reduction annotations to worry about.